expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/effect/PenanceStatusEffect.java

```diff
@@ -0,0 +1,86 @@
+ *
+ * In addition, as a special exception, the copyright holders of
+ * this mod give you permission to combine this mod
+ * with free software programs or libraries that are released under the GNU LGPL
+ * and with code included in the standard release of Minecraft under All Rights Reserved (or
+ * modified versions of such code, with unchanged license).
+ * You may copy and distribute such a system following the terms of the GNU GPL for this mod
+ * and the licenses of the other code concerned.
+ *
+ * Note that people who make modified versions of this mod are not obligated to grant
+ * this special exception for their modified versions; it is their choice whether to do so.
+ * The GNU General Public License gives permission to release a modified version without this exception;
+ * this exception also makes it possible to release a modified version which carries forward this exception.
+ */
+package ladysnake.pandemonium.common.entity.effect;
+
+import ladysnake.pandemonium.Pandemonium;
+import ladysnake.pandemonium.common.PlayerSplitter;
+import ladysnake.pandemonium.common.entity.PlayerShellEntity;
+import ladysnake.requiem.api.v1.event.requiem.PossessionStartCallback;
+import ladysnake.requiem.api.v1.possession.Possessable;
+import ladysnake.requiem.api.v1.possession.PossessionComponent;
+import ladysnake.requiem.api.v1.remnant.RemnantComponent;
+import ladysnake.requiem.api.v1.remnant.RemnantType;
+import ladysnake.requiem.api.v1.remnant.StickyStatusEffect;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.attribute.AttributeContainer;
+import net.minecraft.entity.damage.DamageSource;
+import net.minecraft.entity.effect.StatusEffect;
+import net.minecraft.entity.effect.StatusEffectType;
+import net.minecraft.entity.mob.MobEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.server.network.ServerPlayerEntity;
+import org.jetbrains.annotations.NotNull;
+
+public class PenanceStatusEffect extends StatusEffect {
+    protected PenanceStatusEffect(StatusEffectType type, int color) {
+        super(type, color);
+    }
+
+    @Override
+    public void onApplied(LivingEntity entity, AttributeContainer attributes, int amplifier) {
+        super.onApplied(entity, attributes, amplifier);
+        if (amplifier >= 1) {
+            PossessionComponent c = PossessionComponent.KEY.getNullable(entity);
+            if (c != null) { // It's 1 because amplifiers are 0 based for some fucking reason
+                c.stopPossessing();
+            }
+            if (entity instanceof ServerPlayerEntity && !(RemnantComponent.get((PlayerEntity) entity)).isVagrant()) {
+                if (RemnantComponent.get((PlayerEntity) entity).getRemnantType().isDemon()) {
+                    PlayerSplitter.split((ServerPlayerEntity) entity);
+                } else {
+                    entity.damage(DamageSource.MAGIC, amplifier*4);
+                }
+            }
+        }
```

Comment:
```suggestion
        if (amplifier >= 2 && entity instanceof ServerPlayerEntity) {
            ServerPlayerEntity player = (ServerPlayerEntity) entity;
            if (amplifier >= 3) {
	            PossessionComponent.get(player).stopPossessing();
	        }
            RemnantComponent remnant = RemnantComponent.get(player);
            if (!remnant.isVagrant()) {
                if (remnant.getRemnantType().isDemon()) {
                    PlayerSplitter.split(player);
                } else {
                    player.damage(DamageSource.MAGIC, amplifier*4);
                }
            }
        }
```

Reply:
That's not correct: amplifiers are 0-based, so amplifier >= 2 would be levels higher than 3. Also, the stopping of possession is starting at level 2, not level 3.

Reply:
The stopping at possession is indeed at level 3, not 2, at least in the plans we discussed on discord.
However you're very correct for the first part, I should have read your comment more carefully.